e2e/ansible/playbooks/feature/ha/ha.yml

```diff
@@ -62,40 +78,73 @@
-         register: result
-         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+       - block:
+           - name: Get storage ctrl pod name
+             shell: >
+               source ~/.profile;
+               kubectl get pods -n {{ namespace }} -l openebs/controller=jiva-controller --no-headers
+             args:
+               executable: /bin/bash
+             register: result
+             delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+           - name: Set ctrl pod name to variable
+             set_fact:
+               ctrl_pod_name: "{{ result.stdout.split()[0] }}"
+         when: storage_engine == 'jiva'
 
-       - name: Set ctrl pod name to variable
-         set_fact:
-           ctrl_pod_name: "{{ result.stdout.split()[0] }}"
+       - block:
+           - name: Get storage ctrl pod name
+             shell: >
+               source ~/.profile;
+               kubectl get pods -n {{ operator_ns }} -l openebs.io/controller=cstor-controller | grep {{ namespace }}
+             args:
+               executable: /bin/bash
+             register: result
+             delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+           - name: Set ctrl pod name to variable
+             set_fact:
+               ctrl_pod_name: "{{ result.stdout.split()[0] }}"
+         when: storage_engine == 'cStor'
 
-       - name: Get node on which the pod is scheduled
-         shell: >
-           source ~/.profile;
-           kubectl get pods -n {{ namespace }} -o wide | grep {{ctrl_pod_name}} | awk {'print $7'}
-         args:
-           executable: /bin/bash
-         register: result
-         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+       - block:
+           - name: Get node on which the pod is scheduled
+             shell: >
+               source ~/.profile;
+               kubectl get pods -n {{ namespace }} -o wide | grep {{ctrl_pod_name}} | awk {'print $7'}
+             args:
+               executable: /bin/bash
+             register: result
+             delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+           - name: Store the ctrl pod node name in a variable
+             set_fact:
+               ctrl_node: "{{ result.stdout }}"
+         when: storage_engine == 'jiva'
 
-       - name: Store the ctrl pod node name in a variable
-         set_fact:
-           ctrl_node: "{{ result.stdout }}"
+       - block:
+           - name: Get node on which the pod is scheduled
+             shell: >
+               source ~/.profile;
+               kubectl get pods -n {{ operator_ns }} -o wide | grep {{ctrl_pod_name}} | awk {'print $7'}
+             args:
+               executable: /bin/bash
+             register: result
+             delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+           - name: Store the ctrl pod node name in a variable
+             set_fact:
+               ctrl_node: "{{ result.stdout }}"
+         when: storage_engine == 'cStor'
```

Comment:
@nsathyaseelan , just wanted to understand - are we doing the above (separate blocks for cstor and jiva) only because the namespaces in which the volumes are deployed are different ?